src/Esprima/Ast/NodeList.cs

```diff
@@ -32,6 +42,24 @@ public int Count
         get => _count;
     }
 
+    public T this[int index]
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get
+        {
+            // Following trick can reduce the range check by one
+            if ((uint) index >= (uint) _count)
```

Comment:
nit: I think this is usually written as `(uint) index < (uint) _count` because we want to have hot path as something that matches first

Reply:
I used to make the hot path the first branch until I read or heard somewhere that it doesn't really make a difference with today's CPUs because of branch prediction. Have I got it wrong?

Of course, I'm going to invert it, just being curious.

Reply:
Honestly,  I'm not sure.  It's just nyt gut reaction to try to guide to shortest path.